airflow/models/dataset.py

```diff
@@ -203,6 +206,44 @@ def __repr__(self):
         return f"{self.__class__.__name__}({', '.join(args)})"
 
 
+class DatasetEventManager(LoggingMixin):
+    """
+    A pluggable class that manages operations for dataset events.
+
+    The intent is to have one place to handle all DatasetEvent-related operations, so different
+    Airflow deployments can use plugins that broadcast dataset events to each other.
+    """
+
+    def register_dataset_change(
+        self, *, task_instance=None, dataset=None, session: Session = NEW_SESSION
```

Comment:
should add `extra` to signature here and pass to DatasetEvent

should we consider renaming it `payload`? wdyt @jedcunningham @ashb ?

Reply:
I don't have a strong preference here. I'm okay with both.

Reply:
I like `extra` better, since we're closer to the DB here than we are to an HTTP request.